### PR TITLE
Machine RancherOS Support

### PR DESCRIPTION
**Pull request by wlan0**
@ibuildthecloud  @sheng-liang

### CODE REVIEW

**Comment by wlan0** [comment]
@ibuildthecloud  @sheng-liang -- ACPID daemon does not receive events inside a container

The issue has something to do with the mount namespaces, you can see it running though. 

I've updated the code.

**Comment by wlan0** [comment]
I propose that we do not containerize acpid for now. It is really not required to get rancheros ready for machine. 

We can revisit this, once we have a working machine build.

@ibuildthecloud  @sheng-liang

**Comment by wlan0** [comment]
Also note, this PR requires OS base with acpid support

File: config/default.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ func NewConfig() *Config {
 				Nameservers: []string{"8.8.8.8", "8.8.4.4"},
 			},
 			Interfaces: map[string]InterfaceConfig{
-				"eth0": {
+				"eth*": {
```
Comment (ibuildthecloud) [inline config/default.go]: Why are you doing changing this?

File: init/bootstrap.go [hunk 2]
```diff
@@ -43,10 +46,18 @@ outer:
 			continue
 		}
 
-		for _, b := range buffer {
-			if b != 0 {
-				log.Infof("%s not empty", dev)
-				continue outer
+		boot2docker := false
+
+		if strings.HasPrefix(string(buffer[:30]), boot2dockerMagic) {
```
Comment (ibuildthecloud) [inline init/bootstrap.go]: please do len(b2dmagic) instead of 30

File: config/default.go (hunk 1 above)
Comment (wlan0) [inline config/default.go]: machine for virtualbox configures two interfaces.

File: scripts/dockerimages/scripts/console.sh [hunk 3]
```diff
@@ -33,6 +33,15 @@ if [ ! -d ${RANCHER_HOME} ]; then
...
 fi
 
+DOCKER_HOME=/home/docker
+if [ ! -d ${DOCKER_HOME} ]; then
+    mkdir -p ${DOCKER_HOME}
+    chown docker:docker ${DOCKER_HOME}
+    chmod 2755 ${DOCKER_HOME}
+fi
+
+/usr/sbin/acpid
```
Comment (ibuildthecloud) [inline scripts/dockerimages/scripts/console.sh]: This needs to be a separate container?

Comment (wlan0) [inline scripts/dockerimages/scripts/console.sh]: No, it need not be.

**Pull request merged**